vspace/src/main/java/edu/asu/diging/vspace/core/aspects/previewDataAspect.java

```diff
@@ -0,0 +1,52 @@
+package edu.asu.diging.vspace.core.aspects;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.After;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.ui.Model;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+import org.springframework.web.servlet.HandlerMapping;
+
+import edu.asu.diging.vspace.core.model.impl.Exhibition;
+import edu.asu.diging.vspace.core.services.IExhibitionManager;
+import edu.asu.diging.vspace.web.publicview.ExhibitionConstants;
+
+@Component
+@Aspect
+public class previewDataAspect {
+
+    @Autowired
+    private IExhibitionManager exhibitionManager;
+
+    @Around("execution(public * edu.asu.diging.vspace.web..*Controller.*(..))")
```

Comment:
the preview info only needs to be added to the controllers for the public site, doesn't it? Not for all controllers.

Reply:
yes it is true but Homecontroller also needs to be intercept and the HomeController is in edu.asu.diging.vspace.web not edu.asu.diging.vspace.web.publicview